File: BackEnd/CSharp/ServerParser.cs
using System.Collections.Generic;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace WebApp
{
    static class ServerParser
    {
        public static object ParseServerJS(string location)
        {
            string Server = System.IO.File.ReadAllText(location);
            var Actions = new List<string>();
            var FuncHandler = new ServerSide();
            foreach (System.Reflection.MethodInfo _Method in FuncHandler.GetType().GetMethods())
            {
                if (_Method.DeclaringType.Name != "ServerSide")
                    continue;
                Actions.Add(_Method.Name);
            }

            foreach (var Action in Actions)
            {
                while (Server.Contains(Action + "("))
                {
                    string Paramaters = Generic.GetStringBetween(Server, Action + "(", ")");
                    if (Paramaters == "ERROR")
                        continue;
                    Server = Server.Replace(Action + "(" + Paramaters + ")", Conversions.ToString(Interaction.CallByName(FuncHandler, Action, CallType.Method, Paramaters)));
                }
            }

            return Server;
        }
    }
}

File: BackEnd/VB.NET/ServerParser.vb
Module ServerParser
    Function ParseServerJS(location As String)
        Dim Server As String = IO.File.ReadAllText(location)
        Dim Actions As List(Of String) = New List(Of String)
        Dim FuncHandler As New ServerSide
        For Each _Method As Reflection.MethodInfo In FuncHandler.GetType().GetMethods()
            If _Method.DeclaringType.Name <> "ServerSide" Then Continue For
            Actions.Add(_Method.Name)
        Next
        For Each Action In Actions
            While Server.Contains(Action + "(")
                Dim Paramaters As String = GetStringBetween(Server, Action + "(", ")")
                If Paramaters = "ERROR" Then Continue For
                Server = Server.Replace(Action + "(" + Paramaters + ")", CStr(CallByName(FuncHandler, Action, Microsoft.VisualBasic.CallType.Method, Paramaters)))
            End While
        Next
        Return Server
    End Function
End Module
